### DISCUSSION TITLE
Error in the vs code extension: ECONNREFUSED 127.0.0.1

### DISCUSSION THREAD

**Discussion by f-schnabel** [community/community#51076] [Copilot Conversations] [answered]
Hi, I can't get Copilot to working, it always shows the following error message:
`GitHub Copilot could not connect to server. Extension activation failed: "tunneling socket could not be established, cause=connect ECONNREFUSED 127.0.0.1:80"`
I also looked at other discussions like https://github.com/orgs/community/discussions/43188 and tried to enable the proxy support but it still doesn't work.
I'm also using the Google DNS (8.8.8.8) which was suggested in this discussion with a similar error message: https://github.com/orgs/community/discussions/17413
I have a VPN Client installed (FortiClient version 7.0.3.0193). The problem is the same if I'm connected to a VPN or not.

Does anyone have any idea what I can try?

**Comment by Tanglumy** [comment]
Me,too. Is there any solutions?

**Comment by f-schnabel** [comment, marked as answer]
I discovered the solution that worked for me. It turned out that the `http_proxy` environment variable was set, even though I didn't recall setting it. Once I deleted it, the issue was resolved and everything functioned properly 😄.

**Comment by windead** [comment]
@Shadow-Devil , but I'm truely need to set the **http_proxy** environment, but I'm sure I set it correctly, not the "127.0.0.1:80".

**Comment by Nunchakus888** [comment]
<img width="754" alt="image" src="https://github.com/community/community/assets/14994145/bcb9a073-2eff-4e02-bef4-aa608a27c80a">


default.exp-tas.com
copilot-proxy.githubusercontent.com
origin-tracker.githubusercontent.com
*.githubcopilot.com
github.com
api.github.com/user
api.github.com/copilot_internal/*
default.exp-tas.com
download.jetbrains.com
download-cf.jetbrains.com
cache-redirector.jetbrains.com
plugins.jetbrains.com

**Reply by Nunchakus888** [reply]
<img width="419" alt="image" src="https://github.com/community/community/assets/14994145/fe061e85-ccb0-42d1-9bec-20f75282b38b">

**Reply by hurun** [reply]
It works for me. thanks

**Comment by Zardinality** [comment]
my scenario is a bit different so I will leave my 2 cents. I have to use proxy because I want to connect to a remote server tunnel, so I set the proxy in User settings, to localhost:7890, say. Turns out this setting is also used in remote, so my remote machine also tries to use the proxy server(localhost:7890) to connect copilot. In that case just override the http_proxy to an empty string in the remote setting json would work.
